docs/reference/scripting/common-script-uses.asciidoc

```diff
@@ -0,0 +1,423 @@
+[[common-script-uses]]
+=== Common scripting use cases
+You can write a script to do almost anything, and sometimes, that's
+the trouble. It's challenging to know what's possible with scripts,
+so the following examples address common uses cases where scripts are
+really helpful.
+
+[[scripting-field-extraction]]
+==== Field extraction
+The goal of this use case is simple; you have fields in your data with a bunch of
+information, but you only want to extract pieces and parts.
+
+There are two options at your disposal:
+
+* <<grok-basics,Grok>> uses a pattern like a regular expression that supports
+aliased expressions that you can reuse. Grok sits on top of regular expressions, so
+any regular expressions are valid in grok as well.
+* <<dissect-processor,Dissect>> extracts structured fields out of a single text
+field within a document, but doesn't use regular expressions. Instead, dissect uses
```

Comment:
"extracts structured fields out of text using a pattern that defines delimiters"?

Reply:
How about:
>Dissect extracts structured fields out of text, using delimiters to define the matching pattern. Unlike grok, dissect
doesn't use regular expressions.